giraffe/src/utils/fromFlux.ts

```diff
@@ -189,6 +189,151 @@ export const fromFlux = (fluxCSV: string): FromFluxResult => {
   }
 }
 
+export const fastFromFlux = (fluxCSV: string): FromFluxResult => {
+  const columns: Columns = {}
+  const fluxGroupKeyUnion = new Set<string>()
+  const resultColumnNames = new Set<string>()
+  let tableLength = 0
+
+  try {
+    const chunks = splitChunksFast(fluxCSV)
+
+    // declaring all nested variables here to reduce memory drain
+    let tableText = ''
+    let tableData: any = []
+    let annotationText = ''
+    let columnType: any = ''
+    let columnKey = ''
+    let columnDefault: any = ''
+
+    for (const chunk of chunks) {
```

Comment:
should be operating on indices here and not on the malloc'd chunks. eg: instead of splittedChunk = chunk.split('\n'), use the same iteration technique from the while loop that split up chunks, to split up newlines as well. iirc, because of json and the like, you only want to use this technique for finding the annotation header information before the results, and use papaparse for the rest of the response body